### PR TITLE
Linting messages

### PR DESCRIPTION
**Pull request by mattwigway**
I find I'm all the time pushing code with undefined messages. Find those and complain.

### CODE REVIEW

File: bin/mastarm [hunk 1]
```diff
@@ -102,6 +102,14 @@ commander
   })
 
 commander
+  .command('messages')
```
Comment (evansiroky) [inline bin/mastarm]: maybe we could call this `lint-messages` so it's more obvious what it does

File: lib/lint-messages.js [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+
+const YAML = require('yamljs')
+const { readFileSync } = require('fs')
+const glob = require('glob')
+
+const IMPORT = /import ([a-zA-Z0-9]+) from '[^ ]+\/messages'/
+
+module.exports = function () {
+  // TODO don't hardwire paths
+  const messages = YAML.parse(readFileSync('configurations/default/messages.yml', 'utf8'))
```
Comment (evansiroky) [inline lib/lint-messages.js]: for the location of the `yml` files, you should use the config sent in other places of mastarm.  See for example:  https://github.com/conveyal/mastarm/blob/dev/bin/mastarm#L24

File: lib/lint-messages.js [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+
+  const missing = [].concat(...glob
+    .sync('lib/**/*.js')
```
Comment (evansiroky) [inline lib/lint-messages.js]: this should be configurable.  The root directory may not always be named this.  Also, sometimes there may be multiple root directories.

**Comment by evansiroky** [comment]
Because this code is heavily opinionated regarding the style of importing messages within code, this needs to be explained in the docs.  Since I anticipate that we'll be using this feature heavily in most of our projects, I think it would be a good idea to make tests for this.  And one more request, I'd like the code to exit with 1 if it detects errors.

**Comment by trevorgerhardt** [comment]
Nice work @mattwigway! First question, is a new command necessary for this or would it be better served as part of the `test` pipeline?

File: lib/lint-messages.js (hunk 2 above)
Comment (trevorgerhardt) [inline lib/lint-messages.js]: Yup, see this file also: https://github.com/conveyal/mastarm/blob/dev/lib/load-config.js

**Comment by mattwigway** [comment]
Thanks for the comments! Sorry, I didn't intend to make this seem like it was ready to go. I have another few things to do. I'm thinking this should be part of `mastarm lint`, but to get started it was easier to just have a new command. I'm going to make it less opinionated about how messages are used today.

File: bin/mastarm (hunk 1 above)
Comment (mattwigway) [inline bin/mastarm]: Good call, fixed.

**Comment by mattwigway** [comment]
@trevorgerhardt or @evansiroky I think I fixed everything y'all mentioned, so this is ready for review #staticcodeanalysiswithregularexpressions

File: README.md [hunk 4]
```diff
@@ -171,6 +172,35 @@ Options:
 
 ```
 
+### `lint-messages`
```
Comment (evansiroky) [inline README.md]: Please provide a link to this in the table of contents

File: lib/lint-messages.js [hunk 5]
```diff
@@ -0,0 +1,119 @@
...
+
+    return false
+  })
+}
+
+module.exports = {
+  lint: function (files, messages) {
+    const allFiles = [].concat(
+      ...files.map((file) => {
+        if (statSync(file).isDirectory()) {
```
Comment (evansiroky) [inline lib/lint-messages.js]: Would be good to add a check if the path exists before checking if it is a directory.  A helpful warning message could be displayed if no paths were found.

File: README.md [hunk 6]
```diff
@@ -171,6 +172,35 @@ Options:
...
+
```
Comment (evansiroky) [inline README.md]: For consistency with other sections of this readme we should include the command line options displayed upon running `mastarm lint-messages --help`.

File: __tests__/lib/lint-messages.js [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+  project: {
+    new: 'New'
+  }
+}
+
+describe('lint-messages', () => {
+  const { lintFileContents, parseImportLine } = require('../../lib/lint-messages')
+  it('should parse root import correctly', () => {
+    expect(parseImportLine("import msgs from '../utils/messages'"))
+      .toMatchSnapshot("import msgs from '../utils/messages'")
```
Comment (evansiroky) [inline __tests__/lib/lint-messages.js]: interesting, I didn't know it was possible to name snapshots like this

**Comment by mattwigway** [comment]
@evansiroky thanks for the review, I believe I've addressed your concerns.

File: README.md [hunk 8]
```diff
@@ -16,6 +16,7 @@
   * [Commit](#commit)
   * [Deploy](#deploy)
   * [Lint](#lint)
+  * [Lint Messages](#lint-messages)
```
Comment (evansiroky) [inline README.md]: The order here is inconsistent with the order in the document

**Pull request merged**